lib/ed25519.js

```diff
@@ -0,0 +1,146 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import {
+  sign,
+  verify,
+  createPrivateKey,
+  createPublicKey
+} from 'crypto';
+
+// used to export node's public keys to buffers
+const publicKeyEncoding = {format: 'der', type: 'spki'};
+// used to export node's private keys to buffers
+const privateKeyEncoding = {format: 'der', type: 'pkcs8'};
+// used to turn private key bytes into a buffer in DER format
+const DER_PRIVATE_KEY_PREFIX = Buffer.from(
+  '302e020100300506032b657004220420', 'hex');
+// used to turn public key bytes into a buffer in DER format
+const DER_PUBLIC_KEY_PREFIX = Buffer.from('302a300506032b6570032100', 'hex');
+
+/**
+ * The key material is the part of the buffer after the DER Prefix.
+ *
+ * @param {Buffer} buffer - A DER encoded key buffer.
+ *
+ * @throws {Error} If the buffer does not contain a valid DER Prefix.
+ *
+ * @returns {Buffer} The key material part of the Buffer.
+*/
+function getKeyMaterial(buffer) {
+  if(buffer.indexOf(DER_PUBLIC_KEY_PREFIX) === 0) {
+    return buffer.slice(DER_PUBLIC_KEY_PREFIX.length, buffer.length);
+  }
+  if(buffer.indexOf(DER_PRIVATE_KEY_PREFIX) === 0) {
+    return buffer.slice(DER_PRIVATE_KEY_PREFIX.length, buffer.length);
+  }
+  throw new Error('Expected Buffer to match Ed25519 Public or Private Prefix');
+}
+
+/**
+ * Takes a Buffer or Uint8Array and adds a DER Private Prefix to it.
+ * Allows Uint8Arrays to be interoperable with node's crypto functions.
+ *
+ * @param {object} options - Options to use.
+ * @param {Buffer} [options.privateKeyBytes] - Required if no seedBytes.
+ * @param {Buffer} [options.seedBytes] - Required if no privateKeyBytes.
+ *
+ * @throws {TypeError} Throws if the supplied buffer is not of the right size
+ *  or not a Uint8Array or Buffer.
+ *
+ * @returns {Buffer} DER private key prefix + key bytes.
+*/
+function privateKeyDerEncode({privateKeyBytes, seedBytes}) {
+  if(!(privateKeyBytes || seedBytes)) {
+    throw new TypeError('`privateKeyBytes` or `seedBytes` is required.');
+  }
+  if(!privateKeyBytes && !(seedBytes instanceof Uint8Array &&
+    seedBytes.length === 32)) {
+    throw new TypeError('`seedBytes` must be a 32 byte Buffer.');
+  }
+  if(!seedBytes && !(privateKeyBytes instanceof Uint8Array &&
+    privateKeyBytes.length === 64)) {
+    throw new TypeError('`privateKeyBytes` must be a 64 byte Buffer.');
+  }
+  let p;
+  if(seedBytes) {
+    p = seedBytes;
+  } else {
+    // extract the first 32 bytes of the 64 byte private key representation
+    p = privateKeyBytes.slice(0, 32);
+  }
+  return Buffer.concat([DER_PRIVATE_KEY_PREFIX, Buffer.from(p)]);
```

Comment:
`Buffer.concat` accepts `UInt8Array`s so we don't need the extra `Buffer.from` here or any other place.

```suggestion
  return Buffer.concat([DER_PRIVATE_KEY_PREFIX, p]);
```

Reply:
There were two instances of this corrected here: https://github.com/digitalbazaar/ed25519-verification-key-2020/pull/3/commits/93e0432449d939674f819f917bf7279295309ba5